### PR TITLE
[v1.7.x] Additional option to stop consumer after last message on topic

### PR DESCRIPTION
**Pull request by StounhandJ**
Hi, I really miss such a function, so I decided to try to implement it myself. I am waiting for an answer to how much this option is correct and whether it will be added to the package.

### CODE REVIEW

**Comment by mateusjunges** [comment]
Hi @StounhandJ, can you explain this feature a little bit?

**Comment by StounhandJ** [comment]
As I planned, she should receive messages from the topic until the last one, as soon as the answer comes that the topic is empty, he will stop his work.

**Comment by mateusjunges** [comment]
So if I understood correctly, it should stop the consumer as soon as it detects the end of the topic? If so, I think it is better to rename the method from `isReadingToEnd` to `shouldStopAfterLastMessage` and `withReadingToEnd` to `stopAfterLastMessage`. Also, it will be helpful only when reading a topic which already has some messages, because if I create a new topic, create a consumer with this option and start consuming, as soon as the first message is read, the consumer will stop.

**Comment by StounhandJ** [comment]
That's right. If you think this naming is more correct, then I will rename it. Yes, if you start in a new topic where there are no messages, for example, then he will immediately break through, but I believe in some special tasks it will come in handy.

**Comment by mateusjunges** [comment]
Ok. Yes, please rename it. Also, is it possible to add test?

**Comment by StounhandJ** [comment]
I haven't examined the tests in this project yet, so it may take some time.

**Comment by mateusjunges** [comment]
No problems, on your time.

File: src/Config/Config.php [hunk 1]
```diff
@@ -116,6 +118,11 @@ public function isAutoCommit(): bool
         return $this->autoCommit;
     }
 
+    public function isReadingToEnd(): bool
```
Comment (mateusjunges) [inline src/Config/Config.php]: Please rename it to `shouldStopAfterLastMessage`

File: src/Consumers/ConsumerBuilder.php [hunk 2]
```diff
@@ -354,6 +355,19 @@ public function withBatchReleaseInterval(int $batchReleaseIntervalInMilliseconds
         return $this;
     }
 
+    /**
+     * Enable or disable the read to end option
+     *
+     * @param bool $readingToEnd
+     * @return $this
+     */
+    public function withReadingToEnd(bool $readingToEnd = true): self
```
Comment (mateusjunges) [inline src/Consumers/ConsumerBuilder.php]: Rename this to `stopAfterLastMessage`

**Review by mateusjunges — CHANGES_REQUESTED** [review]

**Comment by StounhandJ** [comment]
There is also a question about the [Makefile](https://github.com/mateusjunges/laravel-kafka/blob/v1.7.x/Makefile#L22-L23), since the assembly for testing inside it is outdated, and specifically this [Dockerfile](https://github.com/mateusjunges/laravel-kafka/blob/v1.7.x/dev/Dockerfile#L1)

**Comment by StounhandJ** [comment]
Unfortunately, I was considering this option. If there is an error in "stopAfterLastMessage", the consume method freezes without the possibility of exiting. May put some expectation in addition to this test.

**Comment by mateusjunges** [comment]
I think it is worth adding this test, at least we are covering the case when everything works as expected.

**Comment by StounhandJ** [comment]
What if we try to [use it](https://phpunit.de/manual/6.5/en/risky-tests.html)?

**Comment by StounhandJ** [comment]
Add annotation @medium

**Comment by StounhandJ** [comment]
Yes, this is exactly the option I was considering. Unfortunately, I can't run the tests locally, so I won't bother configuring it. I described the [problem](https://github.com/mateusjunges/laravel-kafka/pull/103#issuecomment-1145994834) with running tests above.
In docker-compose-test.yaml also crashes the test container at startup.

**Comment by mateusjunges** [comment]
The docker image was missing. I've just merged a PR to fix it.

**Pull request merged**